src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPlugin.java

```diff
@@ -142,6 +145,9 @@ public Object execute(String methodName, Callable executeSqlFunc) throws Excepti
         .getValue();
 
     // use a separate thread to execute method
+    if (monitorService == null) {
```

Comment:
In the constructor, you only call create() if it is enabled
Here, you create it every time it's not null

Does that mean that you should have created it if enabled or disabled previously?

Reply:
For the bottom `execute()`, it's only created if `nativeFailureDetectionEnabled` is enabled & if `monitorService` is null. I decided against creating it regardless of `nativeFailureDetectionEnabled` is enabled or disabled as it seems to be a waste of resources if the user never enables it. I'll be moving the creation of `monitorService` out of constructor and only keeping the creation in `execute()`.

Reply:
I think it is a little weird to initialize the monitor service and execute a method inside the same "execute" method.

I propose using a lazy getter.

Something like
```java
IMonitorService getMonitorService() {
  if (this.monitorService == null) {
     monitorService = monitorServiceInitializer.create(this.log);
  }
  return this.monitorService;
}
```

Although I don't understand why we are moving monitorService out of the constructor?

Reply:
> Although I don't understand why we are moving monitorService out of the constructor?
Moving it out so when the user starts with `nativeFailureDetectionEnabled` disabled/false and later re-enables it during execute(), it can still be created and used.